### ISSUE TITLE
[inversify-express-utils] Support for express.Router options, e.g. mergeParams

### ISSUE THREAD

**Issue by ognus** [inversify/InversifyJS#478]
Express.js Router provides a way to specify some useful options when created
```
var router = express.Router([options]);
```
With defaults being
```
{
  caseSensitive: false,
  mergeParams: false,
  strict: false
}
```

*inversify-express-utils* doesn't currently provide a way to set those options on the `Router`.

Personally I need the `mergeParams` to be set to `true`, to be able to specify params in `@Controller`'s path:
```
@Controller('/parent/:parentId')
...
@Get('/child/:childId')
```

We could add it as a parameter to the `build()` method, e.g.:
```
InversifyExpressServer.build([routerOptions])
```

**I am happy to make a PR if this is something that can be considered useful?**

**Comment by remojansen** [comment]
:+1: I'm happy to accept a PR for this. Here are some recommendations for your PR:

- [The Router instance](https://github.com/inversify/inversify-express-utils/blob/e7532fb8a92961a7e1f239fb5fc522192be5e2ae/src/server.ts#L86) should be a property of the [`InversifyExpressServer` class](https://github.com/inversify/inversify-express-utils/blob/e7532fb8a92961a7e1f239fb5fc522192be5e2ae/src/server.ts#L9)

- The class constructor should allow users to optionally pass a custom router instance:

    ```ts
    constructor(container: inversify.interfaces.Container, router?: express.Router) {
        this.container = container;
        this.router = router || express.Router();
    }
    ```

Thanks a lot for contributing :smile:

**Comment by remojansen** [comment]
> Note: this feature will also fix https://github.com/inversify/inversify-express-utils/issues/476

**Comment by ognus** [comment]
Thanks @remojansen :)

I must be missing something, but how specifying single custom router would solve https://github.com/inversify/inversify-express-utils/issues/476? Shouldn't each controller need it's own custom router for different subdomain?

At the moment there is a new router created for each controller, do you suggest to nest those routers under custom `this.router` like so?
```
controllers.forEach((controller: interfaces.Controller) => {
 
 ...

  if (controllerMetadata && methodMetadata) {
    let router: express.Router = express.Router();

    ...

    this.router.use(controllerMetadata.path, ...controllerMetadata.middleware, router);
  }
}

this.app.use('/', this.router);
```

**Comment by remojansen** [comment]
Hello, sorry for the late reply on this.. I've been very busy but this weekend I was able to implement and release this feature:

```ts
let container = new Container();

let router = express.Router({
    caseSensitive: false,
    mergeParams: false,
    strict: false
});

let server = new InversifyExpressServer(container, router);
```

This feature is available at `inversify-express-utils@3.1.0`

**Comment by ognus** [comment]
@remojansen thanks, appreciate that :) looks good 👍
